ICSharpCode.Decompiler/IL/Instructions/ILFunction.cs

```diff
@@ -73,6 +73,12 @@ partial class ILFunction
 		/// </summary>
 		public BlockContainer DeclarationScope { get; internal set; }
 
+		/// <summary>
+		/// Gets the set of captured variables by this ILFunction.
+		/// </summary>
+		/// <remarks>This is populated by the <see cref="TransformDisplayClassUsage" /> step.</remarks>
+		public HashSet<ILVariable> CapturedVariables { get; } = new HashSet<ILVariable>(ILVariableEqualityComparer.Instance);
```

Comment:
For the most part the ILAst is comparing `ILVariable`s by reference.
`ILVariableEqualityComparer` is only when you want different parts of a split variable to compare equal to another.